File: thisweek_masuipeo2/tyama_codeiq2892.hs
#!/usr/bin/env runghc
import Control.Applicative
import Data.List

split p [] = []
split p s = x : (split p $ dropWhile p $ y)
	where (x,y) = break p s

comb n k = if k<0||n<k then 0 else foldl' (\r i -> (r*(n-i) `div` (i+1))) 1 [0..k-1]

main =
	(map read . (split (== ','))) <$> getLine >>= \[a,b,c] ->
		print $ foldl' (\s i -> s+(comb (a-1) i)*(comb (b-1) (c-i-1))*(comb (c-1) i)) 0 [0..c-1]

File: thisweek_masuipeo2/tyama_codeiq2892.rb
#!/usr/bin/ruby
def comb(n,k)
	return 0 if k<0||n<k
	r=1
	k=n-k if k>n/2
	k.times{|i|
		r=r*(n-i)/(i+1)
	}
	r
end

A,B,C=gets.split(',').map(&:to_i)
p (0..C-1).reduce(0){|s,i|s+comb(A-1,i)*comb(B-1,C-1-i)*comb(C-1,i)}